homeassistant/components/vizio/__init__.py

```diff
@@ -68,4 +84,38 @@ async def async_unload_entry(
         )
     )
 
+    if (
+        len(
+            [
+                entry.entry_id
+                for entry in hass.config_entries.async_entries(DOMAIN)
+                if entry.entry_id != config_entry.entry_id
+            ]
+        )
+        == 0
+    ):
```

Comment:
Don't look at how many entries are there, look at how many entries are loaded. 

Also, no need to check `loaded`, just check `if any(…)`

Reply:
I pushed an update but based on your last sentence I'm not sure if I did what you suggested